sdk/core/core/src/az_curl_slist.c

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_curl_slist.h>
+
+#include <az_action.h>
+#include <az_http_header.h>
+#include <az_http_result.h>
+
+#include <_az_cfg.h>
+
+/**
+ * Appends a zero-terminated string to the given CURL list.
+ */
+AZ_NODISCARD az_result
+az_curl_slist_append(struct curl_slist ** const self, char const * const str) {
+  AZ_CONTRACT_ARG_NOT_NULL(self);
+  AZ_CONTRACT_ARG_NOT_NULL(str);
+
+  struct curl_slist * const p_list = curl_slist_append(*self, str);
+  if (p_list == NULL) {
+    return AZ_HTTP_ERROR_PAL;
```

Comment:
this eats errors. But I'm not sure that's avoidable.

Reply:
That's from CURL docs. They may have additional error info somewhere but I suspect it's always OUT_OF_MEMORY.